### PR TITLE
[Core] Remove array_1d size_t Ctor

### PR DESCRIPTION
**Pull request by sunethwarna**
**Description**
This removes the `std::size_t` Ctor from `array_1d` and all occurances of it.

**Changelog**
- Removes `array_1d::array_1d(std::size_t)` ctor
- Removes all occurances of that same ctor

### CODE REVIEW

**Comment by sunethwarna** [comment]
It is not gonna be easy removing this Ctor, it is used in some places with templates where array_1d is passed as an template argument :/. 

It is used via template in here
https://github.com/KratosMultiphysics/Kratos/blob/a5189a5ed4d9c1d67cd70e0bcb169bcf80e90151/applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp#L278

Where Mathutils have it as following
https://github.com/KratosMultiphysics/Kratos/blob/a5189a5ed4d9c1d67cd70e0bcb169bcf80e90151/kratos/utilities/math_utils.h#L1502-L1518

**Comment by philbucher** [comment]
yes thats what I thougtht... :/

File: kratos/containers/array_1d.h [hunk 1]
```diff
@@ -106,15 +106,6 @@ class	array_1d	: public boost::numeric::ublas::vector_expression< array_1d<T, N>
         //sin esto no funciona en windows!!
         //std::fill (data().begin(), data().end(), value_type	());
     }
-    explicit BOOST_UBLAS_INLINE
-    array_1d (size_type array_size):
-        vector_expression<self_type> ()
-    {
-        //sin esto no funciona en windows!!
-        //std::fill (data().begin(), data().end(), value_type	());
```
Comment (philbucher) [inline kratos/containers/array_1d.h:114]: maybe we can at least bring back this version

and also throw an error in case `array_size` is larger than `N`

**Comment by pooyan-dadvand** [comment]
Now I remember. That was the reason that I had to add this ctor also to AMatrix.

**Comment by pooyan-dadvand** [comment]
After cleaning most of the code from other calls...

File: kratos/containers/array_1d.h (hunk 1 above)
Comment (pooyan-dadvand) [reply kratos/containers/array_1d.h:114]: I am not sure throwing an error is a good idea in release. The overhead of having the possibility would be high here.

For adding as debug error, I have my doubt if the overhead of exception handling mechanism is ommited in the release where the if is false or not.

**Comment by sunethwarna** [comment]
We can close this ryt?

**Comment by pooyan-dadvand** [comment]
Shall we close this issue?

**Comment by sunethwarna** [comment]
Closing for the time being.

**Pull request closed without merging**